app/controllers/application_controller.rb

```diff
@@ -0,0 +1,23 @@
+require 'will_paginate/array'
+class ApplicationController < ActionController::Base
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :exception
+  before_action :require_login
+  before_action :current_user
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  helper_method :current_user
+  # The helper_method line allows us to use @current_user in our view files.
```

Comment:
I'd put `helper_method :current_user` at the top with the `before_action`s since this is acting a bit like a `before_action`, based on your comment.

I'm a but confused as to what `helper_method :current_user` is doing, though, that `before_action :current_user` isn't already doing. My understanding is that `before_action :current_user` is giving you access to @current_user in the view files. I think `helper_method :current_user` is redundant in this case.